File: Data Structures & Algorithms/find-lucky-integer-in-an-array/submission-0.cs
public class Solution {
    public int FindLucky(int[] arr) {
        var obj = new Dictionary<int, int>();

        for (int i = 0; i < arr.Length; i++) {
            if (obj.ContainsKey(arr[i])) {
                obj[arr[i]]++;
            } else {
                obj[arr[i]] = 1;
            }
        }

        int max = -1;
        foreach (var (key, value) in obj) {
            if (key == value && value > max) {
                max = value;
            }
        }

        return max;
    }
}

File: Data Structures & Algorithms/majority-element/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    majorityElement(nums) {
        const map = new Map();
        // let majorityElement = -1;
        // let majorityElementKey = -1;
        // const max = 1_000_000_000;
        // const majorityElementCounterPos = new Array(max).fill(1);
        // const majorityElementCounterNeg = new Array(-1 * max).fill(1);
        // console.log(majorityElementCounterPos);
        // console.log(majorityElementCounterNeg);

        // console.log(arr);
        for (let i = 0; i < nums.length; i++) {
            map.set(nums[i], (map.get(nums[i]) ?? 1) + 1);
        }

        let majorityElement = -1;
        let majorityElementKey = -1;
        map.forEach((value, key) => {
            if (value > majorityElement) {
                majorityElement = value;
                majorityElementKey = key;
            }
        });
        return majorityElementKey;
    }
}
